File: openapi-generator/scalaz/src/main/scala/org/openapitools/client/api/VehicleHarshEventResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import VehicleHarshEventResponse._

case class VehicleHarshEventResponse (
  /* URL for downloading the forward facing video */
  downloadForwardVideoUrl: Option[String],
/* URL for downloading the inward facing video */
  downloadInwardVideoUrl: Option[String],
/* URL for downloading the tracked inward facing video */
  downloadTrackedInwardVideoUrl: Option[String],
/* Type of the harsh event. One of: [Crash, Harsh Acceleration, Harsh Braking, Harsh Turn, ROP Engine, ROP Brake, YC Engine, YC Brake, Harsh Event] */
  harshEventType: String,
/* URL of the associated incident report page */
  incidentReportUrl: String,
/* Whether the driver was deemed distracted during this harsh event */
  isDistracted: Option[Boolean],
location: Option[VehicleHarshEventResponseLocation])

object VehicleHarshEventResponse {
  import DateTimeCodecs._

  implicit val VehicleHarshEventResponseCodecJson: CodecJson[VehicleHarshEventResponse] = CodecJson.derive[VehicleHarshEventResponse]
  implicit val VehicleHarshEventResponseDecoder: EntityDecoder[VehicleHarshEventResponse] = jsonOf[VehicleHarshEventResponse]
  implicit val VehicleHarshEventResponseEncoder: EntityEncoder[VehicleHarshEventResponse] = jsonEncoderOf[VehicleHarshEventResponse]
}

File: openapi-generator/javascript-closure-angular/API/Client/VehicleHarshEventResponse.js
goog.provide('API.Client.VehicleHarshEventResponse');

/**
 * Harsh event details for a vehicle
 * @record
 */
API.Client.VehicleHarshEventResponse = function() {}

/**
 * URL for downloading the forward facing video
 * @type {!string}
 * @export
 */
API.Client.VehicleHarshEventResponse.prototype.downloadForwardVideoUrl;

/**
 * URL for downloading the inward facing video
 * @type {!string}
 * @export
 */
API.Client.VehicleHarshEventResponse.prototype.downloadInwardVideoUrl;

/**
 * URL for downloading the tracked inward facing video
 * @type {!string}
 * @export
 */
API.Client.VehicleHarshEventResponse.prototype.downloadTrackedInwardVideoUrl;

/**
 * Type of the harsh event. One of: [Crash, Harsh Acceleration, Harsh Braking, Harsh Turn, ROP Engine, ROP Brake, YC Engine, YC Brake, Harsh Event]
 * @type {!string}
 * @export
 */
API.Client.VehicleHarshEventResponse.prototype.harshEventType;

/**
 * URL of the associated incident report page
 * @type {!string}
 * @export
 */
API.Client.VehicleHarshEventResponse.prototype.incidentReportUrl;

/**
 * Whether the driver was deemed distracted during this harsh event
 * @type {!boolean}
 * @export
 */
API.Client.VehicleHarshEventResponse.prototype.isDistracted;

/**
 * @type {!API.Client.VehicleHarshEventResponse_location}
 * @export
 */
API.Client.VehicleHarshEventResponse.prototype.location;

